script.js

```diff
@@ -1,4 +1,123 @@
+var gameModeDiceRoll = "game_state_dice_roll";
+var gameStateChooseDiceOrder = "game_state_choose_dice";
+var gameCompareScore = "game_compare_score";
+var currentGameMode = gameModeDiceRoll;
+
+var currentPlayer = 1;
+// var userName = "";
+var allPlayerSCore = [];
+
+//create an array to store the values of the 2 dices for both players
+var currentPlayerRolls = [];
+
+//creating helper function for dice roll
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+//creating helper function to roll 2 dice
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  //after this step, ask player to choose the order
+  console.log(currentPlayerRolls[0]);
+  console.log(currentPlayerRolls[1]);
+  return `Welcome Player ${currentPlayer}! <br><br> You rolled ${currentPlayerRolls[0]} for Dice 1 and ${currentPlayerRolls[1]} for Dice 2. <br><br> Please choose the dice number (either 1 or 2) to be your first digit of your final value.`;
+};
+
+var getPlayerScore = function (playerInput) {
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    return `Error! Please only input "1" or "2" to choose your first digit of your final value. <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]}`;
+  }
+
+  // input == 1
+  if (playerInput == 1) {
+    var playerFinalValue = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+
+  // input == 2
+  if (playerInput == 2) {
+    var playerFinalValue = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
```

Comment:
since we have the above input validation, we do not need to check explicitly for both cases. we can just have a default case and check for the other case

```
var playerFinalValue = Number(
      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1]);

if (playerInput == 2) {
    var playerFinalValue = Number(
      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
    );
  }
```